### PR TITLE
revised running a node

### LINKED ISSUE #438: [DOC] Document how to run a node without nearup
**Is your feature request related to a problem? Please describe.**
To launch a betanet or a testnet node one needs to fetch the proper genesis, and find bootnodes. `nearup` handles that, but nowhere do we document how to do it without nearup.

**Describe the solution you'd like**
https://docs.google.com/document/d/1a-_Gj6Bq1YbJ1sY_zv7irWOHSgdfebJ6KEAaNLgqkYw/edit?usp=sharing

### PR DESCRIPTION
**Pull request by thisisjoshford**
- added @SkidanovAlex's  guide to running a node w/o `nearup`
- clarified node types & roles (non-block producing (regular) vs. block producing (validator))
- added more links for further reference
- renamed `running-testnet.md` to `running-a-node.md` to reflect its actual content
- updated various links to point to `running-a-node.md` 
- created `nodes.md` under `/docs/concepts` and moved information about nodes into it from `running-a-node.md` 
- added more detailed information explaining nodes and their behavior / roles in `nodes.md`

fixes #438

### CODE REVIEW

File: docs/concepts/nodes.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - block producing nodes (aka "validator nodes")
+  - non-block producing nodes (aka "regular nodes")
+
+### Non-block producing node ("regular node")
+
+Anyone can run a "regular node" by following one of the methods mentioned in [this guide](/docs/local-setup/running-a-node). 
+
+You may decide to run a node of your own for a few reasons:
+
+  - To view, process, and validate transactions on `MainNet`, `TestNet` or `BetaNet` (†)
```
Comment (bowenwang1996) [inline docs/concepts/nodes.md:23]: I think some reasons for running a regular node include:
1) serve rpc for your own use cases. This includes transaction submission, queries, view calls etc.
2) to index data from the network (running indexer on top of it for example)

File: docs/concepts/nodes.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Nodes that _can_ produce blocks, also referred to as "[Validator Nodes](/docs/validator/staking-overview)", undergo a more extensive vetting process including [staking](/docs/validator/staking).
+
+To run a block producing node, you must have a validator key and be included among the set of block producers. After each [epoch](/docs/concepts/epoch), "validator nodes" are shuffled and randomly selected for the next epoch. 
+
+See "[How do I become a validator](/docs/validator/validator-faq#how-do-i-become-a-validator)" and [validator selection process](https://nomicon.io/Economics/README.html?validator-selection#validator-selection) for more information.
+
+<blockquote class="warning">
+<strong>Note</strong><br><br>
+
+Non-block producing nodes still validate every block and are very important to the network. This network of nodes, that view all transactions taking place, help us to be certain that the chain is correct and no invalid state transitions / forks are occurring.
```
Comment (bowenwang1996) [inline docs/concepts/nodes.md:43]: Before we have challenges, non block producing nodes cannot really prevent invalid state transition from happening. Even with challenges you have to become a fisherman to challenge.

File: docs/local-setup/running-a-node.md [hunk 3]
```diff
@@ -1,68 +1,64 @@
...
-- To join a network as a validator running a "validator node" (see [staking](/docs/validator/staking))
+3) Download the config file:
+    - TestNet [`config.json`](https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore-deploy/testnet/config.json)
+    - BetaNet [`config.json`](https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore-deploy/betanet/config.json)
+    - MainNet [`config.json`](https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore-deploy/mainnet/config.json)
 
-_( † ) `TestNet` is intended to operate as closely (similarly) to `MainNet`  as possible with only stable releases while `BetaNet` follows a weekly release cycle._
+4) Run the following command replacing the paths & networkId:
+    ```bash
+    neard --home=<absolute_path_to_the_desired_location> init --chain-id=<networkId> --genesis=<absolute_path_to_the_downloaded_genesis_file>
```
Comment (bowenwang1996) [inline docs/local-setup/running-a-node.md:23]: `--genesis` argument only applies to betanet and testnet. For mainnet, genesis is built into the binary

File: docs/local-setup/running-a-node.md [hunk 4]
```diff
@@ -125,13 +120,13 @@ If you are running a validator in production you may find it more efficient to j
...
 
-NB. Please ensure you include the `--release` flag. Omitting this will lead to an unoptimized binary being produced that is too slow for a validator to function effectively.
+**Note**: Please ensure you include the `--release` flag. Omitting this will lead to an unoptimized binary being produced that is too slow for a validator to function effectively.
 
 Finally:
 On MacOS or Linux
 
 ```bash
-nearup betanet --nodocker --binary-path path/to/nearcore/target/release
+nearup betanet --nodocker --binary-path <path_to_nearcore_target_release>
```
Comment (bowenwang1996) [inline docs/local-setup/running-a-node.md:129]: this is outdated. The new version of nearup doesn't have an option to run docker inside. You can run the entire nearup from docker. https://github.com/near/nearup/tree/nearup_v2

File: docs/local-setup/running-a-node.md [hunk 5]
```diff
@@ -1,68 +1,64 @@
...
 
-_(If you prefer to use `TestNet` then just replace `betanet` with `testnet` in the command above)_
 
-
-You will then be prompted for an Account ID. You can leave this empty if you would just like to run a node. Validators should use the account ID of the account you want to stake with. See [staking](/docs/validator/staking) if you would like to become a validator.
+You will then be prompted for an Account ID. You can leave this empty if you would just like to run a regular node. Validators should use the account ID of the account you want to stake with. See [staking](/docs/validator/staking) if you would like to become a validator.
 
 ```text
-Enter your account ID (leave empty if not going to be a validator):
+Enter your account ID: (leave empty if not going to be a validator)
```
Comment (chefsale) [inline docs/local-setup/running-a-node.md:61]: this doesn't exists any more, you have to pass the flag --account-id

File: docs/concepts/nodes.md [hunk 6]
```diff
@@ -11,42 +11,42 @@ NEAR Protocol runs on a collection of publicly maintained computers or "nodes".
...
-  - To join `BetaNet` or `MainNet` as a block producing node, aka "Validator Node" (see "[Running a Validator Node](/docs/validator/staking)")
+  - To join `BetaNet` or `MainNet` as a producing node, aka "Validator Node" (see "[Running a Validator Node](/docs/validator/staking)")
 
   _( † ) `TestNet` is intended to operate as similarly to `MainNet`  as possible with only stable releases, while `BetaNet` follows a weekly release cycle._
 
 _( †† ) `LocalNet` would be the right choice if you prefer to avoid leaking information about your work during the development process since `TestNet` and `BetaNet` are *public* networks. `LocalNet` also gives you total control over accounts, economics, and other factors for more advanced use cases (i.e. making changes to `nearcore`)._
 
 
-### Block producing node ("validator node")
+### Producing node ("validator node")
```
Comment (bowenwang1996) [inline docs/concepts/nodes.md:32]: Why? Producing node makes no sense

Comment (thisisjoshford) [reply docs/concepts/nodes.md:32]: This was discussed in a thread on slack. @SkidanovAlex was concerned that calling a node "block producing" would be incorrect in the future when chunk producing begins. Do you have a suggestion?

File: docs/concepts/nodes.md (hunk 2 above)
Comment (thisisjoshford) [reply docs/concepts/nodes.md]: I should clarify that this is referring to raising alarms within the community, not actually preventing realtime issues.

File: docs/local-setup/running-a-node.md (hunk 3 above)
Comment (thisisjoshford) [reply docs/local-setup/running-a-node.md:23]: This PR was to add running a node without using `nearup` to this document. Existing instructions on running `nearup` were not checked for accuracy. Thank you for the update.

File: docs/local-setup/running-a-node.md (hunk 4 above)
Comment (thisisjoshford) [reply docs/local-setup/running-a-node.md:129]: Same as above... existing instructions on `nearup` were not checked for accuracy. I will carve out some time to run through them and update.

File: docs/concepts/nodes.md (hunk 6 above)
Comment (bowenwang1996) [reply docs/concepts/nodes.md:32]: No but let's not call them "producing node". For now block producing node is fine

**Comment by thisisjoshford** [comment]
closed this stale PR as #739 was recently merged covering this topic.

**Pull request closed without merging**